src/thunderbots/software/ai/navigator/obstacle.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "geom/util.h"
+#include "ai/world/world.h"
+
+constexpr double DEFAULT_AVOID_DIST = 0.15;  // meters
+
+class RobotObstacle
+{
+   public:
+    /**
+     * Constructor.
+     *
+     * @param r Robot to create obstacle from.
+     */
+    RobotObstacle(Robot& r, double avoid_dist);
+
+    /**
+     * Returns the violation distance of the given point inside the obstacle
+     * boundary; 0.0 if no violation.
+     *
+     * @param p point to check if violating
+     *
+     * @return Violation distance, defined as distance from the nearest boundary
+     *         to the given point; 0.0 if no violation occured.
+     */
+    double getViolation(Point& p);
+
+    /**
+     * Returns the closest Point on the obstacle boundary to the given Point that
+     * is not inside the boundary. If the given Point does not violate the boundary,
+     * the given Point will be returned.
+     *
+     * @param p Point to check for
+     * @return A point on the boundary nearest to p if p violates the boundary, otherwise
+     *         returns p
+     */
+    Point getNearestValidPoint(Point& p);
+
+    /**
+     * TODO: This function may have to be rewritten depending on our needs later.
+     * Checks if the given Robot will collide with the obstacle based on the current
+     * velocities.
```

Comment:
nit; but I'd say leave the TODO out, if we need to re-write this then we will, so I don't think having it here adds value (though that's just an opinion)

Reply:
I added that there because I was still really unsure about how it would be used when we port the navigator over. I don't want people to think that this function signature is the final form, and can be completely redone based on what we actually need.

Reply:
Hmm, OK, I don't feel too strongly about this. @MathewMacDougall - thoughts?

Reply:
If you link the ticket for porting the navigator (#23 ) then I'm fine leaving it, since it points to a more actionable item than just "something to do".

Reply:
Linked to #23, fixed